skxray/correlation.py

```diff
@@ -0,0 +1,375 @@
+# STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OTHERWISE) ARISING   #
+# IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE   #
+# POSSIBILITY OF SUCH DAMAGE.                                          #
+########################################################################
+
+"""
+
+This module is for functions specific to time correlation
+
+"""
+
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+import six
+import numpy as np
+import numpy.ma as ma
+import logging
+logger = logging.getLogger(__name__)
+import time
+
+import skxray.core as core
+
+
+def auto_corr(num_levels, num_bufs, indices, img_stack, mask=None):
+    """
+    This module is for one time correlation.
+    The multi-tau correlation scheme was used for
+    finding the lag times (delay times).
+
+    Parameters
+    ----------
+    num_levels : int
+        number of levels of multiple-taus
+
+    num_bufs : int, even
+        number of channels or number of buffers in
+        multiple-taus (must be even)
+
+    indices : ndarray
+        indices of the required region of interest's (roi's)
+        dimensions are: (num_rows, num_cols)
+
+    img_stack : ndarray
+        intensity array of the images
+        dimensions are: (num of images, num_rows, num_cols)
+
+    mask : ndarray, optional
+        mask (eg: dead pixel mask)
+
+    Returns
+    -------
+    g2 : ndarray
+        matrix of one-time correlation
+        shape (num_levels, num_qs)
+
+    lag_steps : ndarray
+        delay or lag steps for the multiple tau analysis
+        shape num_levels
+
+    Notes
+    -----
+    In order to calculate correlations for delays, images must be
+    kept for up to the maximum delay. These are stored in the array
+    buf. This algorithm only keeps number of buffers and delays but
+    several levels of delays number of levels are kept in buf. Each
+    level has twice the delay times of the next lower one. To save
+    needless copying, of cyclic storage of images in buf is used.
+
+    References: text [1]_
+
+    .. [1] D. Lumma, L. B. Lurio, S. G. J. Mochrie and M. Sutton,
+        "Area detector based photon correlation in the regime of
+        short data batches: Data reduction for dynamic x-ray
+        scattering," Rev. Sci. Instrum., vol 70, p 3274-3289, 2000.
+
+    """
+    if num_bufs%2 != 0:
+        raise ValueError("number of channels(number of buffers) in"
+                         " multiple-taus (must be even)")
+
+    if indices.shape == img_stack[0].shape[1:]:
+        raise ValueError("Shape of an image should be equal to"
+                         " shape of the indices array")
+
+    if mask is not None:
+        roi_inds = ma(indices)
+    else:
+        roi_inds = indices
+
+    #  to get indices, number of roi's, number of pixels in each roi's and
+    #  pixels indices for the required roi's.
+    q_inds, num_qs, num_pixels, pixel_list = _get_roi_info(roi_inds)
```

Comment:
Can you rename this to remove 'q'?  Something like `roi_inds, num_rois, num_pixels, pixel_list = _get_roi_info(...)`